File: 050-059/ABC052_B.js
// ABC052 B - Increment Decrement
'use strict'

function Main(input) {
    const tmp = input.trim().split('\n');
    const N = Number(tmp[0])
    const S = tmp[1].split('')
    // console.log(N, S);

    let result = 0;
    let x = 0;
    for (const v of S) {
        if (v === 'I') {
            x++;
        } else if (v === 'D') {
            x--;
        }
        if (x > result) {
            result = x;
        }
    }
    
    console.log(result);
}

Main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: 050-059/ABC052_B.c
// ABC052 B - Increment Decrement
#include <stdio.h>

int main(void) {
    int N;
    char S[101];
    // input
    scanf("%d", &N);
    scanf("%s", S);

    int x = 0;
    int result = 0;
    int i = 0;
    while (S[i] != '\0') {
        if (S[i] == 'I') {
            x++;
        } else {
            x--;
        }
        if (x > result) {
            result = x;
        }
        i++;
    }

    printf("%d\n", result);
}